universal_transfer_operator/transfers/data_providers/base.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import Any
+
+from airflow.hooks.base import BaseHook
+from datasets.base import UniversalDataset as Dataset
+
+from constants import FileLocation, LoadExistStrategy
+from utils import get_dataset_connection_type
+
+
+class DataProviders(ABC):
+    """
+    Base class to represent all the DataProviders interactions with Dataset.
+
+    The goal is to be able to support new dataset by adding
+    a new module to the `uto/data_providers` directory, without the need of
+    changing other modules and classes.
+    """
+
+    def __init__(
+        self,
+        dataset: Dataset,
+        transfer_mode,
+        transfer_params: dict = {},
+        if_exists: LoadExistStrategy = "replace",
+    ):
+        self.dataset = dataset
+        self.transfer_params = transfer_params
+        self.transfer_mode = transfer_mode
+        self.if_exists = if_exists
+        self.transfer_mapping: Any = {}
```

Comment:
Can you add a comment on what this is for? I can't recall

Reply:
[Done](https://github.com/astronomer/astro-sdk/pull/1492/commits/831d627d1a86acab300968fc56d11bc7f016b4b0)